src/bwamem.cpp

```
@@ -1239,6 +1247,19 @@ void mem_chain_seeds(FMI_search *fmi, const mem_opt_t *opt,
             assert(sa_coord != NULL);
         }
+
+        /* Phase A buffer (Spec S4): ceiling = (#SMEMs this read) * max_occ, the
+         * same prefetch ceiling sa_coord is sized to. Grow with realloc, persist
+         * across reads. Reset nrec per read. */
+        int64_t nrec = 0;
+        int64_t recs_need = (pos - smem_ptr + 1) * (int64_t)opt->max_occ;
+        if (recs_need > recs_cap)
+        {
+            recs_cap = recs_need;
+            recs = (seed_rec_t *) realloc(recs, recs_cap * sizeof(seed_rec_t));
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)